File: Firebase/creative_api/functions/src/Service/VehicleService.js
const Vehicle = require("./../Class/Vehicle");
const vehicleRepository = require("./../Repository/VehicleRepository");

class VehicleSevice {
    async saveVehicle(vehicle) {
        return (await vehicleRepository.save(vehicle.get()));
    }
    async getById(id) {
        return (await vehicleRepository.find(id));
    }
    async getAll() {
        return (await vehicleRepository.findAll());
    }
    async updateVehicle(vehicle){
        return (await vehicleRepository.update(vehicle.get()));
    }
    async findByLicensePlate(license_plate) {
        return (await vehicleRepository.findByLicensePlate(license_plate));
    }
}

module.exports = new VehicleSevice();

File: classes PI/Truckleer/Truckleer/Modules/Vehicle/VehicleService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Truckleer.Modules
{
    class VehicleService
    {
        //Vehicle repository property
        readonly VehicleRepository repository;
        //Class constructor
        public VehicleService()
        {
            //initializate vehicle repository property
            repository = new VehicleRepository();
        }
        //Method to findAll vehicles
        public List<Vehicle> FindAll()
        {
            return repository.FindAll().Result;
        }
        //Method to find one specific vehicle by id
        public Vehicle FindOne(string Id)
        {
            return repository.Find(Id).Result;
        }
        //Method to Save or Update vehicle
        public bool Save(Vehicle vehicle)
        {
            return repository.Save(vehicle).Result;
        }
    }
}
